tests/examples/shims/screen.lua

```diff
@@ -85,8 +87,36 @@ function screen._get_extents()
     return xmax.geometry.x+xmax.geometry.width, ymax.geometry.y+ymax.geometry.height
 end
 
+-- The way the `screen` module is used to store both number and object is
+-- problematic since it can cause invalid screens to be "resurrected".
+local function catch_invalid(_, s)
```

Comment:
I do not understand this method. It is equivalent to the following (minus producing different error messages):
```lua
local function catch_invalid(_, s)
    if s == nil or type(s) == "string" then return end
    error("Well, error") -- The only other return (type(s) == "number") is after an assert(type(s) == "table") and thus is dead code.
end
```

Reply:
>  (minus producing different error messages):

That's the whole point